source/module_lr/potentials/xc_kernel.cpp

```diff
@@ -163,105 +165,83 @@ void LR::KernelXC::f_xc_libxc(const int& nspin, const double& omega, const doubl
-            //     // 2. $\nabla^2(f^{\sigma\sigma}*\sigma)$
-            //     std::vector<double> v2sigma2_sigma_r(3 * nrxx);
-            //     for (int ir = 0; ir < nrxx; ++ir)
-            //     {
-            //         v2sigma2_sigma_r[ir] = v2s2.at(ir * 6) * sigma[ir * 3] * 4.0
-            //             + v2s2.at(ir * 6 + 1) * sigma[ir * 3 + 1] * 4.0
-            //             + v2s2.at(ir * 6 + 3) * sigma[ir * 3 + 2];   //up-up
-            //         v2sigma2_sigma_r[nrxx + ir] = v2s2.at(ir * 6 + 1) * sigma[ir * 3] * 2.0
-            //             + v2s2.at(ir * 6 + 4) * sigma[ir * 3 + 2] * 2.0
-            //             + (v2s2.at(ir * 6 + 2) * 4.0 + v2s2.at(ir * 6 + 3)) * sigma[ir * 3 + 1];   //up-down
-            //         v2sigma2_sigma_r[2 * nrxx + ir] = v2s2.at(ir * 6 + 5) * sigma[ir * 3 + 2] * 4.0
-            //             + v2s2.at(ir * 6 + 4) * sigma[ir * 3 + 1] * 4.0
-            //             + v2s2.at(ir * 6 + 3) * sigma[ir * 3];   //down-down
-            //     }
-            //     for (int isig = 0;isig < 3;++isig)
-            //         LR_Util::lapl(v2sigma2_sigma_r.data() + isig * nrxx, v2sigma2_sigma_r.data() + isig * nrxx, *(chg_gs.rhopw), tpiba2);
-            //     // 3. $\nabla^2(v^\sigma)$
-            //     std::vector<double> lap_vsigma(3 * nrxx);
-            //     for (int ir = 0;ir < nrxx;++ir)
-            //     {
-            //         lap_vsigma[ir] = vs.at(ir * 3) * 2.0;
-            //         lap_vsigma[nrxx + ir] = vs.at(ir * 3 + 1);
-            //         lap_vsigma[2 * nrxx + ir] = vs.at(ir * 3 + 2) * 2.0;
-            //     }
-            //     for (int isig = 0;isig < 3;++isig)
-            //         LR_Util::lapl(lap_vsigma.data() + isig * nrxx, lap_vsigma.data() + isig * nrxx, *(chg_gs.rhopw), tpiba2);
-            //     // add to v2rho2_
-            //     BlasConnector::axpy(3 * nrxx, 1.0, v2r2.data(), 1, to_mul_rho_.data(), 1);
-            //     BlasConnector::axpy(3 * nrxx, -1.0, div_v2rhosigma_gdrho_r.data(), 1, to_mul_rho_.data(), 1);
-            //     BlasConnector::axpy(3 * nrxx, 1.0, v2sigma2_sigma_r.data(), 1, to_mul_rho_.data(), 1);
-            //     BlasConnector::axpy(3 * nrxx, 1.0, lap_vsigma.data(), 1, to_mul_rho_.data(), 1);
-            // }
-            else
+        // add onto the total components
+        auto omp_add_vector = [](const std::vector<double>& src, std::vector<double>& dst)
             {
-                throw std::domain_error("nspin =" + std::to_string(nspin)
-                    + " unfinished in " + std::string(__FILE__) + " line " + std::to_string(__LINE__));
-            }
-        }
+                assert(src.size() == dst.size());
+#ifdef _OPENMP
+#pragma omp parallel for 
+#endif
+                for (size_t i = 0; i < src.size(); ++i) { dst[i] += src[i]; }
+            };
+        omp_add_vector(vrho_tmp, this->vrho_);
```

Comment:
Have you tested the performance of doing these vector add in one OMP block or do these one by one?

Reply:
> Have you tested the performance of doing these vector add in one OMP block or do these one by one?

Here's the performance test (by the code commented out in commit 7855e113) with an example of nrxx=16777216 on my 32-core machine:

chunk_size | 1 | 64 | 1024 | 4096 | (nrxx/32+1) | Not specify | dynamic | serial (no omp)
-- | -- | -- | -- | -- | -- | -- | -- | -- 
time(ms) | 189 | 76 | 61 | 60 | 65 | 67 | 173 | 109

So finally I chose `schedule(static, 4096)` here.